worlds/yugioh06/opponents.py

```diff
@@ -0,0 +1,257 @@
+        OpponentData(9, "Water Dragon", ["Quick-Finish"], 2, 5, 6481, 8009, "deck/LV2_waterD.ydc\x00\x00\x00", 1),
+        # Tier 3
+        OpponentData(10, "Red-Eyes Darkness Dragon", ["Quick-Finish"], 3, 1, 6292, 8010, "deck/LV3_RedEyes.ydc\x00\x00",
+                     2),
+        OpponentData(11, "Vampire Genesis", ["Quick-Finish"], 3, 2, 6293, 8011, "deck/LV3_vamp.ydc\x00\x00\x00\x00\x00",
+                     2),
+        OpponentData(12, "Infernal Flame Emperor", [], 3, 3, 6368, 8012, "deck/LV3_flame.ydc\x00\x00\x00\x00", 2,
+                     ["Has Back-row removal"]),
+        OpponentData(13, "Ocean Dragon Lord - Neo-Daedalus", [], 3, 4, 6376, 8013, "deck/LV3_daidaros.ydc\x00", 2,
+                     ["Has Back-row removal"]),
+        OpponentData(14, "Helios Duo Megiste", ["Quick-Finish"], 3, 5, 6647, 8014, "deck/LV3_heriosu.ydc\x00\x00", 2),
+        # Tier 4
+        OpponentData(15, "Gilford the Legend", ["Quick-Finish"], 4, 1, 6456, 8015, "deck/LV4_gilfo.ydc\x00\x00\x00\x00",
+                     3),
+        OpponentData(16, "Dark Eradicator Warlock", ["Quick-Finish"], 4, 2, 6530, 8016, "deck/LV4_kuromadou.ydc", 3),
+        OpponentData(17, "Guardian Exode", [], 4, 3, 6640, 8017, "deck/LV4_exodo.ydc\x00\x00\x00\x00", 3),
+        OpponentData(18, "Goldd, Wu-Lord of Dark World", ["Quick-Finish"], 4, 4, 6505, 8018, "deck/LV4_ankokukai.ydc",
+                     3),
+        OpponentData(19, "Elemental Hero Erikshieler", ["Quick-Finish"], 4, 5, 6639, 8019,
+                     "deck/LV4_Ehero.ydc\x00\x00\x00\x00", 3),
+        # Tier 5
+        OpponentData(20, "Raviel, Lord of Phantasms", [], 5, 1, 6565, 8020, "deck/LV5_ravieru.ydc\x00\x00", 4),
+        OpponentData(21, "Horus the Black Flame Dragon LV8", [], 5, 2, 6100, 8021, "deck/LV5_horus.ydc\x00\x00\x00\x00",
+                     4),
+        OpponentData(22, "Stronghold", [], 5, 3, 6153, 8022, "deck/LV5_gadget.ydc\x00\x00\x00", 5),
+        OpponentData(23, "Sacred Phoenix of Nephthys", [], 5, 4, 6236, 8023, "deck/LV5_nephthys.ydc\x00", 6),
+        OpponentData(24, "Cyber End Dragon", ["Goal"], 5, 5, 6397, 8024, "deck/LV5_cyber.ydc\x00\x00\x00\x00", 7),
+    ]
+    world = multiworld.worlds[player]
+    if not randomize:
+        return opponents_table
+    opponents = opponents_table + challenge_opponents
+    start = world.random.choice([o for o in opponents if o.tier == 1 and len(o.additional_info) == 0])
+    opponents.remove(start)
+    goal = world.random.choice([o for o in opponents if "Goal" in o.campaign_info])
+    opponents.remove(goal)
+    world.random.shuffle(opponents)
+    chosen_ones = opponents[:23]
+    for item in (multiworld.precollected_items[player]):
+        if item.name in tier_1_opponents:
+            # convert item index to opponent index
+            chosen_ones.insert(item_to_index[item.name] - item_to_index["Campaign Tier 1 Column 1"], start)
+            continue
```

Comment:
I'm not 100% sure that I understand this part of the code - Can I get a quick explanation?

Reply:
This should've been a break instead of continue